cmake/posix/px4_impl_posix.cmake

```diff
@@ -196,8 +196,17 @@ function(px4_os_add_flags)
 				)
 		endif()
 
-	else()
+	elseif(CYGWIN)
+		set(added_definitions
+			-D__PX4_POSIX
+			-D__PX4_CYGWIN
+			-D_GNU_SOURCE
```

Comment:
Do you need this and if so why?

Reply:
Itz is needed because:
- for example: https://github.com/PX4/Firmware/blob/fa5010109edf0c71914fc04bb2dc887918336f65/src/drivers/device/posix/cdev_platform.cpp#L334 uses `pthread_getname_np` and also in other places of the code `pthread_setname_np` is used if I remember correctly.
- Cygwin uses newlib c library and after clean installation in /usr/include/pthread.h there is this section:
```
/* Non posix calls */

#if __GNU_VISIBLE
int pthread_getattr_np (pthread_t, pthread_attr_t *);
int pthread_getname_np (pthread_t, char *, size_t) __attribute__((__nonnull__(2)));
int pthread_setname_np (pthread_t, const char *) __attribute__((__nonnull__(2)));
int pthread_sigqueue (pthread_t *, int, const union sigval);
int pthread_yield (void);
#endif
#if __MISC_VISIBLE /* HP-UX, others? */
int pthread_suspend (pthread_t);
int pthread_continue (pthread_t);
#endif
```
- I found out by reading this discussion: https://github.com/magnumripper/JohnTheRipper/issues/1250#issuecomment-101845784 that `_GNU_SOURCE` is the define to use for enabling the build configuration that internally includes the define `__GNU_VISIBLE`
- I added it
- It works

If you have any better solution please feel free to correct. Honestly I don't understand all the interna about these libraries and in my opinion it's so hard to find any documentation.

Reply:
Interesting and thanks for looking it up. I'm wondering why we don't need to define it then (or do I see that wrong)?
In any case I would try to reduce the scope to only the files where `pthread_{g,s}etname_np` is used. The define can lead to subtle changes in behavior.
Something like:
```
#define _GNU_SOURCE // for pthread_getname_np
#include <pthread.h>
#undef _GNU_SOURCE
```

Reply:
I see that `_GNU_SOURCE` is actually defined in our build, probably because we add `-std=gnu++11`.
So you can ignore my comment and it's ok to add the define globally (would be good to make this more explicit though for the other builds).

Reply:
> I see that _GNU_SOURCE is actually defined in our build

Why do I have to redefine it then? Is it dependent on the compiler? Does `-std=gnu++11` just not get handeled in cygwin? That would explain it. In any case I don't see a risk for this define.